cmd/minishift/cmd/root.go

```diff
@@ -333,3 +366,16 @@ func setDefaultActiveProfile() {
 		}
 	}
 }
+
+// checkForValidProfileOrExit checks if a profile exist or not when --profile flag used.
+// If profile not exist then it will error out with message.
+func checkForValidProfileOrExit(cmd *cobra.Command) {
+	if cmd.Parent() != nil {
+		// This condition true for each command execpt `minishift profile <subcommand>` and `minishift start ...``
+		if cmd.Parent().Name() != profileFlag && cmd.Name() != startFlag {
```

Comment:
We need to take profile aliases in to account i.e. instance, profiles.

Reply:
Ya forgot about it.

Reply:
@LalatenduMohanty We don't need it here because this handled in the `initializeProfile()` method. Do try once and let me know if you still think it needed.

```
$ minishift instance/profiles set/list/delete 

=> then checkForValidProfileOrExit() function will check first it profile flag which is not then it will check IsvalidProfile which either doesn't exist and it exit then return.
```

Reply:
@praveenkumar we need to fix the logic of this function depending upon what we plan to do with initializeProfile function.